### DISCUSSION TITLE
File names

### DISCUSSION THREAD

**Discussion by mja8020** [mja8020/templar#13] [Ideas]
Would we stick with the following?

templar.yaml in root
project.yaml for folder overrides
.templates folders for template sources

Should we allow .templates to be located in any project folder?

**Comment by terrabyteian** [comment]
Project.yaml feels maybe too generic? Like it could conflict with actual project code. 

What about

Root: templar.yaml/yml
Subdirs: .templar.yaml/yml
Templates: .templar/...

Makes it more clear it's specific to the tool but keeps the files dotted so they show up at the top of most dir trees.

**Reply by terrabyteian** [reply]
Oh and yea am a fan of allowing nested templates

**Comment by mja8020** [comment]
I'm thinking do we even need to differentiate the root project anymore? Maybe just ".templar.yaml" all the way down and we use the top most one as the root?

Also we put templates in ".templates" or use ".templar" as the folder to keep it consistent?

**Reply by terrabyteian** [reply]
Yea we use .templar/* as the template folder

And we do need a root one still I think. There may be intermediate layers of your tree structure that don't have templar customizations per se. 

Just let templar.yaml be the root and .templar.yaml be for any nested customizations

**Reply by terrabyteian** [reply]
Although I could still see a use for not wanting the root... 

Maybe we keep root but let it be optional to climb to, like you can run in a subsection of your repo using the .templar.yaml and optional go all the way to the top

**Reply by terrabyteian** [reply]
Or maybe we just use git root as the top of a repo? Could see all those options

**Reply by mja8020** [reply]
I think using the repo root may be a bad idea.  Lets say you are in a mono repo you may have different folders you want to run this in?  

Maybe we do keep the root templar.yaml and then folders have a .templar.yaml.  I think even if in the root you could have a .templar.yaml.  The idea being something like the following

templar.yaml
```
---
values:
  group: {} # Assuming we have settings or validation for values options would go here (different topic)
  owner: {}
```

Then running `templar init` you get prompted for group and owner values or something like `templar init --values group=foo` the output of which would get stored in the .templar.yaml file.

If that makes sense.

**Reply by mja8020** [reply]
This would prevent running `templar init` from messing with the templay.yaml in the root

**Reply by terrabyteian** [reply]
yea thats a better pattern, lets do that
